instrumentation/opentelemetry-instrumentation-pika/tests/test_utils.py

```diff
@@ -284,3 +331,54 @@ def test_decorate_basic_publish_no_properties(
         get_span.return_value.is_recording.assert_called_once()
         inject.assert_called_once_with(basic_properties.return_value.headers)
         self.assertEqual(retval, callback.return_value)
+
+    @mock.patch("opentelemetry.instrumentation.pika.utils._get_span")
+    @mock.patch("opentelemetry.propagate.inject")
+    @mock.patch("opentelemetry.context.get_current")
+    @mock.patch("opentelemetry.trace.use_span")
+    def test_decorate_basic_publish_with_hook(
+        self,
+        use_span: mock.MagicMock,
+        get_current: mock.MagicMock,
+        inject: mock.MagicMock,
+        get_span: mock.MagicMock,
+    ) -> None:
+        callback = mock.MagicMock()
+        tracer = mock.MagicMock()
+        channel = mock.MagicMock(spec=Channel)
+        method = mock.MagicMock(spec=Basic.Deliver)
+        mock_properties = mock.MagicMock()
+        mock_body = b"mock_body"
+
+        def publish_hook(span, body, properties):
```

Comment:
Why not use `mock.MagicMock()`? This seems like what you want. Same for `test_decorate_callback_with_hook`

Reply:
You can use `mock_callback.assert_called_once_with(...)`